.github/scripts/lib/lesson-promotion-core.ps1

```
@@ -0,0 +1,796 @@
+    $exhibits = @(Get-LPField $m 'exhibits' @())
+    $declared = @(Get-LPField $m 'declared_states' @())
+    $snapshot = Get-LPField $m 'roster_snapshot'
+    $floor = Get-LPField $m 'specificity_floor'
+    $lensFloor = Get-LPField $m 'lens_body_floor'
+    $loadingSurfaces = Get-LPField $m 'loading_surfaces'
+    $pins = @(Get-LPField $m 'in_file_pins' @())
+
+    foreach ($req in @('entries', 'declared_states', 'roster_snapshot', 'specificity_floor', 'lens_body_floor', 'loading_surfaces', 'in_file_pins', 'exhibits', 'schema_version')) {
+        if ($null -eq (Get-LPField $m $req)) { $drift.Add("manifest carries no '$req'; a manifest missing a top-level key must read as drift, not as nothing to check") }
+    }
+    $schemaVersion = [int](Get-LPField $m 'schema_version' 0)
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Get-LPField' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)